org.eclipse.jdt.ls.start/src/org/eclipse/jdt/ls/start/JavaLanguageServerStart.java

```diff
@@ -0,0 +1,144 @@
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.start;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Plugin;
+import org.eclipse.core.runtime.Status;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.SubstituteLoggerFactory;
+
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.joran.JoranConfigurator;
+import ch.qos.logback.core.joran.spi.JoranException;
+import ch.qos.logback.core.util.StatusPrinter;
+
+/**
+ * Override logback preferences *before* M2E plugin is activated below
+ *
+ * @author snjeza
+ *
+ */
+public class JavaLanguageServerStart extends Plugin {
+
+	private static final String LOGBACK_CONFIG_FILE_PROPERTY = "logback.configurationFile";
+	private static final String LOGBACK_DEFAULT_FILENAME = "logback.xml";
+	private static final String LOGBACK_EMPTY_FILENAME = "emptylogback.xml";
+	public static final String PLUGIN_ID = "org.eclipse.jdt.ls.start";
+	private static final String CORE_PLUGIN_ID = "org.eclipse.jdt.ls.core";
+	private static BundleContext context;
+
+	static BundleContext getContext() {
+		return context;
+	}
+
+	@Override
+	public void start(BundleContext bundleContext) throws Exception {
+		super.start(bundleContext);
+		JavaLanguageServerStart.context = bundleContext;
+		if (System.getProperty(LOGBACK_CONFIG_FILE_PROPERTY) == null) {
+			File stateDir = getStateLocation().toFile();
+			boolean isDebug = Boolean.getBoolean("jdt.ls.debug");
+			String fileName = isDebug ? LOGBACK_DEFAULT_FILENAME : LOGBACK_EMPTY_FILENAME;
+			File configFile = new File(stateDir, fileName);
+			if (!configFile.isFile()) {
+				try (InputStream is = bundleContext.getBundle().getEntry(fileName).openStream(); FileOutputStream fos = new FileOutputStream(configFile)) {
+					Files.copy(is, configFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+				} catch (Exception e) {
+					logException(e.getMessage(), e);
+				}
+			}
+			loadConfiguration(configFile.toURI().toURL());
+			// ch.qos.logback.classic.util.ContextInitializer.CONFIG_FILE_PROPERTY
+			// System.setProperty(LOGBACK_CONFIG_FILE_PROPERTY, configFile.getAbsolutePath());
+		}
+		try {
+			Platform.getBundle(CORE_PLUGIN_ID).start(Bundle.START_TRANSIENT);
+		} catch (BundleException e) {
+			logException(e.getMessage(), e);
+		}
```

Comment:
Why do we need to start jdt.ls.core bundle manually here? It looks unnecessary for me.

Reply:
You are right. I have removed it.